vm/systemSmartContracts/governance.go

```diff
@@ -232,10 +264,24 @@ func (g *governanceContract) proposal(args *vmcommon.ContractCallInput) vmcommon
 		g.eei.AddReturnMessage("invalid number of arguments, expected 3")
 		return vmcommon.FunctionWrongSignature
 	}
-	if !g.isWhiteListed(args.CallerAddr) {
-		g.eei.AddReturnMessage("called address is not whiteListed")
+	generalConfig, err := g.getConfig()
```

Comment:
Gas was checked first - so it does not matter. left as it is.

Reply:
Agree on one hand - that by having `UseGas` above the user pays upfront for the whole execution (even if only couple of instructions get executed in case of wrongly passed parameters). 
But we could have avoided get/save config if a user sends a "stupid" transaction. It would be like some extra "breathing" time for the node.

Reply:
this is executed super fast. Only a read from trie.